src/main/java/org/mskcc/oncokb/controller/MongoController.java

```diff
@@ -0,0 +1,341 @@
+                jsonObject.put("BIRTH_DATE", clinical.getBirthDate());
+                jsonObject.put("DFCI_MRN", clinical.getMrn());
+                jsonObject.put("GENDER", clinical.getGender());
+                clinicalArray.put(jsonObject);
+            }
+
+            for(Genomic genomic: annotatedGenomics){
+                JSONObject jsonObject = new JSONObject();
+                jsonObject.put("ONCOKB_GENOMIC_ID", genomic.getGenomicId());
+                jsonObject.put("SAMPLE_ID", genomic.getSampleId());
+                jsonObject.put("TRUE_HUGO_SYMBOL", genomic.getTrueHugoSymbol());
+                jsonObject.put("TRUE_PROTEIN_CHANGE", genomic.getTrueProteinChange());
+                jsonObject.put("TRUE_VARIANT_CLASSIFICATION", genomic.getTrueVariantClassification());
+                jsonObject.put("VARIANT_CATEGORY", genomic.getVariantCategory());
+                jsonObject.put("CNV_CALL", genomic.getCnvCall());
+                jsonObject.put("WILDTYPE", genomic.getWildtype());
+                jsonObject.put("CHROMOSOME", genomic.getChromosome());
+                jsonObject.put("POSITION", genomic.getPosition());
+                jsonObject.put("TRUE_CDNA_CHANGE", genomic.getTrueCdnaChange());
+                jsonObject.put("REFERENCE_ALLELE", genomic.getReferenceAllele());
+                jsonObject.put("TRUE_TRANSCRIPT_EXON", genomic.getTrueTranscriptExon());
+                jsonObject.put("CANONICAL_STRAND", genomic.getCanonicalStand());
+                jsonObject.put("ALLELE_FRACTION", genomic.getAlleleFraction());
+                jsonObject.put("TIER", genomic.getTier());
+                jsonObject.put("ONCOKB_VARIANT",genomic.getOncokbVariant().toString());
+                genomicArray.put(jsonObject);
+            }
+
+            // check if any trials matched for query data
+            matchedResults = findMatchedTrials(previousMatchedRecordsSet, genomicArray, clinicalArray);
+
+            // drop collection "trial_query" first to clean records for previous queries
+            // create a new collection "trial_query" to save matched trials.
+            Boolean isDropped = MongoUtil.dropCollection(this.mongoDatabase, "trial_query");
+            if (matchedResults.size() > 0) {
+                if(isDropped) {
+                    Boolean isCreated = MongoUtil.createCollection(this.mongoDatabase,
+                        "trial_query", new ArrayList<>(matchedResults));
+                    if (!isCreated) {
+                        return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+                }
+            }
+
+            String clinicalPath = FileUtil.buildJsonTempFile(clinicalArray.toString(), clinicalTempFile);
+            String genomicPath = FileUtil.buildJsonTempFile(genomicArray.toString(), genomicTempFile);
+
+            ProcessBuilder loadPb = new ProcessBuilder("python", System.getenv("CATALINA_HOME") +
```

Comment:
What happened if `CATALINA_HOME ` is not available?

Reply:
![image](https://user-images.githubusercontent.com/14971266/37434363-b22e4f7c-27b5-11e8-995f-534a9646a0fc.png)
I will add `matchengine.absolute-path` in application.yml, so developers can fill the matchengine folder path mannully. Program will get the path first then get $CATALINA_HOME if the path is null.